src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs

```diff
@@ -294,30 +306,32 @@ public static MediaTypeSegmentWithQuality CreateMediaTypeSegmentWithQuality(stri
             var parsedMediaType = new MediaType(mediaType, start, length: null);
             var parser = parsedMediaType._parameterParser;
 
+            // ??? In every other ParseNextParameter() loop, we treat a parameter as valid as long as it comes before
+            // ??? anything invalid in the media type string. Why is this case different? Should the MediaType
+            // ??? constructor validate the parameters, allowing this loop to exit early and increasing consistency?
             double quality = 1.0d;
             MediaTypeParameter parameter;
             while (parser.ParseNextParameter(out parameter))
             {
                 if (parameter.HasName(QualityParameter))
                 {
+                    // ??? This code does not detect the error if a media type contains 2 q parameters. File a bug?
```

Comment:
Well,

1. Everything falls over due to the comments you've included in the media type 😈 
2. The case I raised was `text/html;charset=utf-8;q=0.9;q=0.8` or `text/html;charset=utf-8;q=0.9;s=23;t=24;q=0.8`. Today that results in `quality=0.8` which is incorrect.
3. We can't exit the loop early because this is the only place in the code that checks `ParsingFailed`. That flag isn't very useful unless the `ParseNextParameter()` loop completes normally.